### ISSUE TITLE
[BUG] Error thrown when navigating between videos

### ISSUE THREAD

**Issue by czengg** [jhomlala/betterplayer#464]
**Describe the bug**
I am currently using betterplayer to build an instagram stories-like experience, using urls hosted in cloudfront. Occassionally when I navigate between videos I get this error:
`AVAudioSession_iOS.mm:1149  Deactivating an audio session that has running I/O. All I/O should be stopped or paused prior to deactivating the audio session.`

**Example code*
```
class VideoPlayerBackground extends StatefulWidget {
  const VideoPlayerBackground(
      {required this.key,
      this.url,
      this.muted = false,
      this.loop = false,
      this.overlay = false,
      required this.child,
      this.onEnd,
      this.filePath,
      this.hideControls = false,
      this.controller,
      this.assetPath})
      : super(key: key);

  final Key key;
  final String? url;
  final bool muted;
  final bool loop;
  final bool overlay;
  final Widget child;
  final Function? onEnd;
  final String? filePath;
  final String? assetPath;
  final bool hideControls;
  final VideoPlayerBackgroundController? controller;

  @override
  _VideoPlayerBackgroundState createState() => _VideoPlayerBackgroundState();
}

class _VideoPlayerBackgroundState extends State<VideoPlayerBackground> {
  bool _visible = false;
  Timer? _timer;

  late VideoPlayerBackgroundController _controller;

  late StreamSubscription _endedSubscription;

  @override
  void initState() {
    super.initState();

    if (widget.controller != null) {
      _controller = widget.controller!;
    } else {
      _controller = VideoPlayerBackgroundController(
          muted: widget.muted,
          loop: widget.loop,
          url: widget.url,
          path: widget.filePath);
    }

    WidgetsFlutterBinding.ensureInitialized();
    SystemChrome.setPreferredOrientations([
      DeviceOrientation.portraitUp,
    ]);
  }

  @override
  void didChangeDependencies() {
    super.didChangeDependencies();

    if (_controller.controller == null ||
        !(_controller.controller!.isVideoInitialized() ?? false)) {
      _controller.initializeController(context);
    }

    _endedSubscription = _controller.endedStream.listen((event) {
      if (widget.onEnd != null && event) {
        widget.onEnd!();
      }
    });
  }

  @override
  void dispose() {
    super.dispose();

    _endedSubscription.cancel();
    if (_controller.controller != null) {
      _controller.controller!.pause();
    }

    if (_timer != null) {
      _timer!.cancel();
    }
  }

  @override
  Widget build(BuildContext context) {
    return StreamBuilder(
        stream: _controller.controllerUpdateStream,
        builder: (context, snapshot) {
          if (_controller.controller == null) {
            return LoadingIndicator();
          }

          return VisibilityDetector(
              key: Key("video-background-${widget.url}"),
              child: Container(
                  child: Stack(children: [
                LoadingIndicator(),
                Container(
                    height: screenSize(context).height,
                    width: screenSize(context).width,
                    child: BetterPlayer(
                      controller: _controller.controller!,
                    )),
                if (widget.overlay)
                  Align(
                      alignment: Alignment.bottomCenter,
                      child: Container(
                          height: 350,
                          decoration: BoxDecoration(
                              gradient: LinearGradient(
                                  begin: Alignment.bottomCenter,
                                  end: Alignment.topCenter,
                                  colors: [
                                Color.fromRGBO(0, 0, 0, .75),
                                Colors.transparent
                              ])))),
                widget.child,
                if (!widget.hideControls)
                  Visibility(
                      visible: _visible,
                      child: Center(
                          child: Icon(
                        _controller.controller!.isPlaying() ?? false
                            ? Icons.play_arrow
                            : Icons.pause,
                      ))),
              ])),
              onVisibilityChanged: (VisibilityInfo info) {
                if (info.visibleFraction > .8) {
                  _controller.controller!.play();
                } else if (mounted) {
                  _controller.controller!.pause();
                }
              });
        });
  }
}
```

**Better Player version**
- Version: ^0.0.65

**Smartphone (please complete the following information):**
 - Device: iPhone 12 pro
 - OS: iOS 14.4.2

**Comment by jhomlala** [comment]
@czengg This will be fixed in next release. Thanks for example but it's not runnable. Please attach minimal reproducible example, not whole solution from your app.

**Comment by czengg** [comment]
@jhomlala Thanks for the response! I was able to fix by changing the way I render the player completely, but good to know that this will be fixed in next release. Would you still like an example that's runnable for reference? I can try to pull up my previous code.

**Comment by jhomlala** [comment]
Released in 0.0.66.
